vsphere_plugin_common/__init__.py

```diff
@@ -154,6 +155,91 @@ def connect(self, cfg):
         username = cfg['username']
         password = cfg['password']
         port = cfg['port']
+        certificate_path = cfg.get('certificate_path')
+        # Until the next major release this will have limited effect, but is
+        # in place to allow a clear path to the next release for users
+        allow_insecure = cfg.get('allow_insecure', False)
+
+        if certificate_path and allow_insecure:
+            raise NonRecoverableError(
+                'Cannot connect when certificate_path and allow_insecure '
+                'are both set. Unable to determine whether connection should '
+                'be secure or insecure.'
+            )
+        elif certificate_path:
+            if not hasattr(ssl, '_create_default_https_context'):
+                raise NonRecoverableError(
+                    'Cannot create secure connection with this version of '
+                    'python. This functionality requires at least python '
+                    '2.7.9 and has been confirmed to work on at least 2.7.12.'
+                )
+
+            if not os.path.exists(certificate_path):
```

Comment:
It wouldn't, but I wanted to make sure that we gave useful output if a user pointed at a directory full of certs (since some things support that, but this doesn't). Without the separate checks we'd have to give the same error message in either error case (something like 'does not exist or is a directory'), which I think is a bit less user friendly.